### ISSUE TITLE
Array issue

### ISSUE THREAD

**Issue by amrlafi** [esiegel/rqlGenerator#1]
The following RQL renders a few issues with the grammer:
`in(php,engines) & in(xslt, php.extensions) & php.version > 4.1.0 & php.version < 5.0 & (os.Type = Linux | os.type = FeeBSD) & disk.space >= 20000 & memory >= 40960`

I received the following errors:
line 1:7 mismatched input 'engines' expecting VALUE_ARRAY
line 1:27 mismatched input 'php' expecting VALUE_ARRAY
line 1:45 mismatched input 'php' expecting {'and', 'or', 'count', 'distinct', 'eq', 'first', 'ge', 'gt', 'in', 'le', 'limit', 'ne', 'select', 'sort', '('}
line 1:67 mismatched input 'php' expecting {'and', 'or', 'count', 'distinct', 'eq', 'first', 'ge', 'gt', 'in', 'le', 'limit', 'ne', 'select', 'sort', '('}
line 1:88 mismatched input 'os' expecting {'count', 'distinct', 'eq', 'first', 'ge', 'gt', 'in', 'le', 'limit', 'ne', 'select', 'sort'}
line 1:106 mismatched input 'os' expecting {'and', 'or', 'count', 'distinct', 'eq', 'first', 'ge', 'gt', 'in', 'le', 'limit', 'ne', 'select', 'sort', '('}
line 1:126 mismatched input 'disk' expecting {'and', 'or', 'count', 'distinct', 'eq', 'first', 'ge', 'gt', 'in', 'le', 'limit', 'ne', 'select', 'sort', '('}
line 1:148 mismatched input 'memory' expecting {'and', 'or', 'count', 'distinct', 'eq', 'first', 'ge', 'gt', 'in', 'le', 'limit', 'ne', 'select', 'sort', '('}

**Comment by esiegel** [comment]
I'm sorry to say that this wasn't a direct/exact port of the original library, and to be honest I wrote this so long ago that I don't remember much of the details.

Looking at the grammar it seems to be incomplete.
- It does not support infix notation `version < 1` and requires `lt(version,1)` for example.
- when using in you need to supply an array with brackets `[1,2,3]` though there seems to be a small bug where commas are missing (i'll fix that). 
- Ids with periods like `php.extentions` are not allowed, but I will update this.

Basically the query would become something like this:

```
in(php,["engines"])&in(xslt,["php.extensions"])&gt(php.version,"4.1.0")&lt(php.version,"5.0")&or(eq(os.Type,"Linux"),(eq(os.type,"FeeBSD")))&ge(disk.space,20000)&ge(memory,40960)
```

Just remember that this code is has been forgotten.

**Comment by amrlafi** [comment]
Thanks for the follow up. I started working on an updated version, I can post it here later.
According to the latest RQL [documentation](https://doc.apsstandard.org/6.0/spec/rql/), string values does not required double quotes.
For example: in(name,(Silver,Gold)) 
The parser gets confused as both Silver and Gold looks like IDs. 

Also consider the following example: in(name,(Joe Nick)), a space becomes more challenging.
